### PR TITLE
created test and api for getApi

### PR DESCRIPTION
**Pull request by silvester-zogaj**

### CODE REVIEW

File: api/app.js [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
-const { getCategories } = require("./controllers/getCategories.controller");
+const { getCategories, getApi } = require("./controllers/getCategories.controller");
+const fs = require('fs/promises');
+
 
 app.get("/api/categories", getCategories);
 
 
+app.get('/api', (request, response) => {
+    fs.readFile(`endpoints.json`, 'utf-8').then((data) => {
```
Comment (JVal-NC) [inline api/app.js]: I would probably sill have this as a controller in a seperate file.

File: __tests__/getCategories.test.js [hunk 2]
```diff
@@ -27,5 +27,20 @@ describe("GET - getCategories", () => {
     })
   });
 
+  describe('GET - endpoints.json', () => {
+    test('should return contents of endpoints.json file ', () => {
+        return request(app)
+        .get('/api')
+        .expect(200)
+        .then((response) => {
+            const {endpoints} = response.body
```
Comment (JVal-NC) [inline __tests__/getCategories.test.js]: Good thinking but I would just require it at the top of the test file and assert against that.

File: api/models/selectCategories.model.js [hunk 3]
```diff
@@ -11,4 +12,12 @@ exports.selectCategories = () => {
     .then((response) => {
         return response.rows
     });
-};
+}
+
+// exports.selectApi = () => {
```
Comment (JVal-NC) [inline api/models/selectCategories.model.js]: Remove commented out code in PRs please

**Review by JVal-NC — COMMENTED** [review]
Couple comments to include. A heads up, having a controller for each endpoint might mean you end up with with lots of files. May want to rationalise these a bit.

**Comment by JVal-NC** [comment]
Comments to include, good work so far.

File: api/models/categories.model.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      `
+    SELECT * from categories
+    `
+    )
+    .then((response) => {
+        return response.rows
+    });
+}
+
+exports.selectApi = () => {
```
Comment (pippaamy) [inline api/models/categories.model.js]: we don't need a model for task 3.5 - we can pop this functionality in the controller :)

**Review by pippaamy — COMMENTED** [review]
great job, lets just pop that functionality in the model and then resubmit :)

**Review by pippaamy — APPROVED** [review]
fab, well done on those changes! go ahead and merge!

**Pull request merged**